commit b48e9f5b169957edca1d05581011a34bb31c16fb (refs/changes/15/1515/4)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-02-15T19:57:27+00:00 (2 years, 8 months ago)
    
    CNAPI-693 CNAPI crashes when Moray returns an initial empty set of servers for allocation
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/lib/common.js b/lib/common.js
index b4135a2..7fd600c 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 var fs = require('fs');
@@ -131,7 +131,7 @@ function getAdminIp(sysinfo) {
 
 function randSort(array) {
     /* Durstenfeld shuffle */
-    for (var i = array.length - 1; i !== 0; i--) {
+    for (var i = array.length - 1; i > 0; i--) {
         var j = Math.floor(Math.random() * (i + 1));
 
         var tmp = array[i];
